### PR TITLE
Capt 1571/extract form object for qualification

### PR DESCRIPTION
**Pull request by rjlynch**
This pr has two separate commits which should be reviewed separately. I'm happy
to break the commits out into separate PRs if that's better.

The main changes are:
* [break out a form object for the qualification step](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/2669/commits/1ca0169730481701342df62518a7443bbd29dd6a)
* [remove a callback from the eligibility model](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/2669/commits/d122746167954e3285dfa41506f7ddbcdf9a53e7)

There's a little more details in the commit message.
I'm not familiar enough to know all the implications of removing the callback from the eligibility model but from what I can tell it seems safe todo and the test suite passes, let me know if I'm wrong and I can drop that commit.

walk-through

### CODE REVIEW

File: app/views/additional_payments/claims/qualification.html.erb
Comment (your) [inline app/views/additional_payments/claims/qualification.html.erb:1]: I just realised that Ken, Abi and I have been moving translation entries under `<journey>.forms.<slug>.questions`.

File: app/forms/journeys/additional_payments_for_teaching/qualification_form.rb
Comment (rjlynch) [reply app/forms/journeys/additional_payments_for_teaching/qualification_form.rb:59]: 👍

File: app/forms/journeys/additional_payments_for_teaching/qualification_form.rb
Comment (rjlynch) [reply app/forms/journeys/additional_payments_for_teaching/qualification_form.rb:53]: Yeah I think that makes sense 👍

File: app/views/additional_payments/claims/qualification.html.erb
Comment (rjlynch) [reply app/views/additional_payments/claims/qualification.html.erb:1]: 👍

File: app/forms/journeys/additional_payments_for_teaching/qualification_form.rb
Comment (your) [reply app/forms/journeys/additional_payments_for_teaching/qualification_form.rb]: Thanks, I think it's what we are doing in the [controller](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/fb4c8e1ec3fd21b19aa4433aaa3f568d8b766ce3/app/controllers/claims_controller.rb#L170-L173), so it's worth having the same default behaviour unless we need really to override it in the subclass (not the norm). This would also fix it for all the new forms (that are failing to show the backlink if submitted but an error occurs)

What do you think about a small test in the base class? Smth like:

```ruby
...
  let(:params) { ActionController::Parameters.new({journey: "test-journey", slug: "test_slug", claim: claim_params}) }
...
  describe "#backlink_path" do
    before do
      allow_any_instance_of(Journeys::PageSequence).to receive(:previous_slug)
        .and_return(previous_slug)
    end

    describe "when the previous slug is present" do
      let(:previous_slug) { "previous-slug" }

      it { expect(form.backlink_path).to eq("/test-journey/previous-slug") }
    end

    describe "when the previous slug is not present" do
      let(:previous_slug) { nil }

      it { expect(form.backlink_path).to be_nil }
    end
  end
```

File: app/forms/journeys/additional_payments_for_teaching/qualification_form.rb
Comment (your) [reply app/forms/journeys/additional_payments_for_teaching/qualification_form.rb:37]: Yes, it's probably worth leaving it for another day.

**Review by your — APPROVED** [review]
No blockers from me, thanks for tackling this. Might be worth waiting for another review as well

**Pull request merged**
